tests/opentelemetry-test-utils/src/opentelemetry/test/test_base.py

```diff
@@ -85,6 +90,61 @@ def sorted_spans(self, spans):  # pylint: disable=R0201
             reverse=True,
         )
 
+    def sorted_metrics(self, metrics):  # pylint: disable=R0201
+        """
+        Sorts metrics by metric name.
+        """
+        return sorted(
+            metrics,
+            key=lambda m: m.name,
+        )
+
+    def get_sorted_metrics(self):
+        resource_metrics = (
+            self.memory_metrics_reader.get_metrics_data().resource_metrics
+        )
+        for metrics in resource_metrics:
+            for scope_metrics in metrics.scope_metrics:
+                all_metrics = list(scope_metrics.metrics)
+                return self.sorted_metrics(all_metrics)
+
+    def assert_metric_expected(
+        self, metric, expected_value, expected_attributes
+    ):
+        data_point = next(iter(metric.data.data_points))
+
+        if isinstance(data_point, HistogramDataPoint):
+            self.assertEqual(
+                data_point.sum,
+                expected_value,
+            )
+        elif isinstance(data_point, NumberDataPoint):
+            self.assertEqual(
+                data_point.value,
+                expected_value,
+            )
+
+        self.assertDictEqual(
+            expected_attributes,
+            dict(data_point.attributes),
+        )
+
+    def assert_duration_metric_expected(
+        self, metric, duration_estimated, expected_attributes, est_delta=200
+    ):
+        data_point = next(iter(metric.data.data_points))
+
+        self.assertAlmostEqual(
+            data_point.sum,
+            duration_estimated,
+            delta=est_delta,
+        )
```

Comment:
This could just be call to `assert_metric_expected`

Reply:
This function checks assertAlmostEqual, for cases like server/client duration.
so there is an extra argument for delta.